### ISSUE TITLE
image_dataset_from_directory returns dtype 'uint8' Dataset if interpolation='nearest'

### ISSUE THREAD

**Issue by buoi** [tensorflow/tensorflow#49829]
## URL(s) with the issue:
https://www.tensorflow.org/api_docs/python/tf/keras/preprocessing/image_dataset_from_directory

### Clear description
tf.keras.preprocessing.image_dataset_from_directory returns dtype 'uint8' Dataset if interpolation='nearest'
however function docstring reports misleading:

### Returns defined

Returns:    A `tf.data.Dataset` object.      - If `label_mode` is None, it yields `float32` tensors of shape        `(batch_size, image_size[0], image_size[1], num_channels)`,        encoding images (see below for rules regarding `num_channels`).      - Otherwise, it yields a tuple `(images, labels)`, where `images`        has shape `(batch_size, image_size[0], image_size[1], num_channels)`,        and `labels` follows the format described below.

[...]
## Description of issue (what needs changing):

return dtype 'uint8' for interpolation='nearest should be specified, such as:

Returns:    A `tf.data.Dataset` object.      - If `label_mode` is None, it yields `float32` tensors of shape        `(batch_size, image_size[0], image_size[1], num_channels)`,        encoding images (see below for rules regarding `num_channels`).     
 - If `label_mode` is None and interpolation is 'bilinear' it yields `uint8`.
 - Otherwise, it yields a tuple `(images, labels)`, where `images`        has shape `(batch_size, image_size[0], image_size[1], num_channels)`,        and `labels` follows the format described below.

[...]

**Comment by Saduf2019** [comment]
Thank you for the information, Pr is created to track this issue #49861

**Comment by Saduf2019** [comment]
As per comment on  #49861, this issue is moved to keras repo, hence moving this to closed status, requesting you to follow up at [this link](https://github.com/keras-team/keras/pull/15429)
